### PR TITLE
docs: move database model to mermaid

### PR DESCRIPTION
**Pull request by ndom91**
## Reasoning 💡

1. Migrated the png database model to dynamically generated Mermaid diagram. This does not need any svg copying back n forth though, thankfully. The inline mermaid "code" just gets built into an inline svg at build time automatically.

A little modification had to be done to have the text color show up correctly in dark mode, but its not so bad.

2. I also added a small bit of text to each Adapter's (which had its own "Schema" section) page that notes that its schema is based upon our "master schema/model" and linked to the `/adapters/model` page and its new mermaid schema.

3. Updated three `@docusaurus/*` dependencies to their latest versions.

Resurrected this PR: https://github.com/nextauthjs/docs/pull/182

## Checklist 🧢

- [x] Documentation
- [ ] Tests
- [ ] Ready to be merged

## Affected issues 🎟

image

image

### CODE REVIEW

**Comment by ndom91** [comment]
@balazsorban44 fixed this to the new docs path

**Comment by ndom91** [comment]
@balazsorban44 I think this is good to go. What do you think?

File: docs/docs/adapters/dgraph.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ The quickest way to use Dgraph is by applying the unsecure schema to your [local
 This approach is not secure or for production use, and does not require a `jwtSecret`.
 :::
 
+> This schema is adapted for use in Dgraph and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/dgraph.md:48]:
```suggestion
> This schema is adapted for use in Dgraph and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/dynamodb.md [hunk 2]
```diff
@@ -77,22 +77,22 @@ The table respects the single table design pattern. This has many advantages:
 - Querying relations is faster than with multi-table schemas (for eg. retrieving all sessions for a user).
 - Only one table needs to be replicated, if you want to go multi-region.
 
-Here is the default schema of the table:
+> This schema is adapted for use in DynamoDB and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/dynamodb.md:80]:
```suggestion
> This schema is adapted for use in DynamoDB and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/fauna.md [hunk 3]
```diff
@@ -80,3 +80,5 @@ CreateIndex({
   terms: [{ field: ["data", "identifier"] }, { field: ["data", "token"] }],
 })
 ```
+
+> This schema is adapted for use in FaunaDB and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/fauna.md:84]:
```suggestion
> This schema is adapted for use in Fauna and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/mikro-orm.md [hunk 4]
```diff
@@ -33,6 +33,8 @@ export default NextAuth({
 
 The MikroORM adapter ships with its own set of entities. If you'd like to override them, you can optionally pass them to the adapter.
 
+> This schema is adapted for use in MikroORM and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/mikro-orm.md:36]:
```suggestion
> This schema is adapted for use in MikroORM and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/models.md [hunk 5]
```diff
@@ -3,9 +3,47 @@ id: models
 title: Models
 ---
 
-NextAuth.js can be used with any database. Models tell you what structures NextAuth.js expects from your database. Models will vary slightly depending on which adapter you use, but in general will look something like this.
-
-![v4 Schema](/img/nextauth_v4_schema.png)
+NextAuth.js can be used with any database. Models tell you what structures NextAuth.js expects from your database. Models will vary slightly depending on which adapter you use, but in general will look something like this. Each adapter's model / schema will be slightly adapted for its needs, but will look very much like this schema below:
```
Comment (balazsorban44) [inline docs/docs/adapters/models.md:6]:
```suggestion
NextAuth.js can be used with any database. Models tell you what structures NextAuth.js expects from your database. Models will vary slightly depending on which adapter you use, but in general, will look something like this. Each adapter's model/schema will be slightly adapted for its needs, but will look very much like this schema below:
```

File: docs/docs/adapters/neo4j.md [hunk 6]
```diff
@@ -58,7 +58,7 @@ The following relationships and relationship labels are used.
 
 ### Properties
 
-See [models](/adapters/models) for the node properties. Relationships have no properties.
+This schema is adapter for use in Neo4J and is based upon our master [models](/adapters/models). Please check there for the node properties. Relationships have no properties.
```
Comment (balazsorban44) [inline docs/docs/adapters/neo4j.md:61]:
```suggestion
This schema is adapted for use in Neo4J and is based upon our main [models](/adapters/models). Please check there for the node properties. Relationships have no properties.
```

File: docs/docs/adapters/prisma.md [hunk 7]
```diff
@@ -41,6 +41,8 @@ Schema for the Prisma Adapter (`@next-auth/prisma-adapter`)
 
 You need to use at least Prisma 2.26.0. Create a schema file in `prisma/schema.prisma` similar to this one:
 
+> This schema is adapted for use in Prisma and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/prisma.md:44]:
```suggestion
> This schema is adapted for use in Prisma and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/sequelize.md [hunk 8]
```diff
@@ -42,6 +42,8 @@ export default NextAuth({
 
 By default, the sequelize adapter will not create tables in your database. In production, best practice is to create the [required tables](https://next-auth.js.org/adapters/models) in your database via [migrations](https://sequelize.org/master/manual/migrations.html). In development, you are able to call [`sequelize.sync()`](https://sequelize.org/master/manual/model-basics.html#model-synchronization) to have sequelize create the necessary tables, foreign keys and indexes:
 
+> This schema is adapted for use in Sequelize and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/sequelize.md:45]:
```suggestion
> This schema is adapted for use in Sequelize and based upon our main [schema](/adapters/models)
```

File: docs/docs/adapters/typeorm.md [hunk 9]
```diff
@@ -44,7 +44,9 @@ The TypeORM adapter uses [`Entity` classes](https://github.com/typeorm/typeorm/b
 
 If you want to override the default entities (for example to add a `role` field to your `UserEntity`), you will have to do the following:
 
-1. create a file containing your modified entities:
+> This schema is adapted for use in TypeORM and based upon our master [schema](/adapters/models)
```
Comment (balazsorban44) [inline docs/docs/adapters/typeorm.md:47]:
```suggestion
> This schema is adapted for use in TypeORM and based upon our main [schema](/adapters/models)
```

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**